File: Valt/Compiler/Lex/Token.cs
namespace Valt.Compiler.Lex
{
    public struct Token
    {
        public string Text;
        public TokenType Type;
        public override string ToString()
        {
            if (Type == TokenType.Eoln)
            {
                return $"'' ({Type})";
            }
            return $"'{Text}' ({Type})";
        }
    }
}

File: JavaPort/src/lexing/Token.kt
package lexing

enum class TokenType{
    Identifier,
    Number,
    Spaces,
    Eoln,
    Quote,
    Reserved,
    Operator,
    Comment,
    SharpPragmaOrInclude,
    Directive,
}
data class Token(val text: String, val kind: TokenType)
